packages/actions/types.ts

```diff
@@ -38,7 +39,17 @@ export interface ActionsProps {
   size?: 'small' | 'medium'
   /**
    * Array of CTAs. Minimum of one, max of two.
-   * @see CtaProps
-   */
-  ctas: [CtaProps] | [CtaProps, CtaProps]
+   * When both CtaProps and StandaloneLinkProps are
+   * provided, they are only accepted in that order.
+   * This is intentional and ensures adherence to a design
+   * requirement of rendering the Button first when data
+   * for both a Button and a StandaloneLink are provided.
+   * @see CtaProps and StandaloneLinkProps
+   */
+  ctas:
+    | [CtaProps]
+    | [StandaloneLinkProps]
+    | [CtaProps, CtaProps]
+    | [StandaloneLinkProps, StandaloneLinkProps]
+    | [CtaProps, StandaloneLinkProps]
```

Comment:
We enforce the order of the data here and thus the order of the CTAs.

Reply:
I know the underlying components differ in props they accept, but I think when composing them we can make them the same.

Example:

```tsx
<Actions
  ctas={[
    {
      title: 'One',
      href: '/one'
    },
    {
      type: 'standalone-link',
      title: 'Two',
      href: '/two'
    }
  ]}
/>
```

CTA type:

```tsx
ctas: Array<{
  type: 'button' | 'standalone-link'
  title: string
  href: string
}>
```

and we default type to button.

This will remove the need to do the isButton and isStandalone link checks and just rely on the type passed in.

Thoughts?

Reply:
If I'm understanding correctly, yes I think this makes way more sense.

Is [this](https://github.com/hashicorp/react-components/pull/758/commits/f8d2272227df856ddd2eaf965f8ca4af40f718da) what you had in mind?

Reply:
Yeah, I think we can update CtaProps to remove variant.

```ts
interface CtaProps {
  /**
   * The text that appears inside the button.
   */
  title: string
  /**
   * Where the button links to when clicked.
   */
  href: string
  /**
   * A function that will be called when the button is clicked.
   */
  onClick?: () => void
}

interface ButtonCtaProps extends CtaProps {
  type?: 'button'
}

interface StandaloneLinkCtaProps extends CtaProps {
  type: 'standalone-link'
}
```

Reply:
I'd prefer we use href vs url since it matches the native link attribute. eg <a href="">...</a>.

In the future it'd be nice to move from url usage in button to href as well.

Reply:
> I'd prefer we use href vs url since it matches the native link attribute. eg <a href="">...</a>.

Totally onboard. Wouldn't changing `url` to `href` on `CtaProps` require either changes to models that are supplying data to `Actions` or data transformation after we query Dato? If so, do you think that is something that should be tackled in this PR? Or are you just flagging this more generally?

Reply:
For clarity I am not saying updating button prop right now. But action ctas can use href vs url and we just pass the href to the url on button.